From db2c3e6f0ff5fe84783227a1df3a1944c7667f19 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 25 Dec 2014 11:07:47 +0800
Subject: [PATCH 418/602] MA-6062 Add selinux config for dnsmasq to access
 prop net.dnsmasq.completed

WFD Sink application count on net.dnsmasq.completed to check the dns result

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 evk_6sl/BoardConfig.mk       | 3 ++-
 imx6/sepolicy/dnsmasq.te     | 6 ++++++
 sabreauto_6q/BoardConfig.mk  | 3 ++-
 sabreauto_6sx/BoardConfig.mk | 3 ++-
 sabresd_6dq/BoardConfig.mk   | 3 ++-
 sabresd_6sx/BoardConfig.mk   | 3 ++-
 6 files changed, 16 insertions(+), 5 deletions(-)
 create mode 100644 imx6/sepolicy/dnsmasq.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 0fc8de7..39275fb 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -97,4 +97,5 @@ BOARD_SEPOLICY_UNION := \
        rild.te \
        init.te \
        netd.te \
-       bootanim.te  
+       bootanim.te \
+       dnsmasq.te
diff --git a/imx6/sepolicy/dnsmasq.te b/imx6/sepolicy/dnsmasq.te
new file mode 100644
index 0000000..5c8f411
--- /dev/null
+++ b/imx6/sepolicy/dnsmasq.te
@@ -0,0 +1,6 @@
+allow dnsmasq property_socket:sock_file rw_file_perms;
+allow dnsmasq properties_device:file rwx_file_perms;
+allow dnsmasq init:unix_stream_socket connectto;
+allow dnsmasq default_prop:property_service set;
+allow dnsmasq net_radio_prop:property_service set;
+
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 3182ac3..1b41b17 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -122,5 +122,6 @@ BOARD_SEPOLICY_UNION := \
        rild.te \
        init.te \
        netd.te \
-       bootanim.te
+       bootanim.te \
+       dnsmasq.te
 
diff --git a/sabreauto_6sx/BoardConfig.mk b/sabreauto_6sx/BoardConfig.mk
index 9257349..c8be829 100755
--- a/sabreauto_6sx/BoardConfig.mk
+++ b/sabreauto_6sx/BoardConfig.mk
@@ -116,5 +116,6 @@ BOARD_SEPOLICY_UNION := \
        rild.te \
        init.te \
        netd.te \
-       bootanim.te
+       bootanim.te \
+       dnsmasq.te
 
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index e4a589d..20ae14a 100644
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -127,6 +127,7 @@ BOARD_SEPOLICY_UNION := \
        rild.te \
        init.te \
        netd.te \
-       bootanim.te
+       bootanim.te \
+       dnsmasq.te
 
 
diff --git a/sabresd_6sx/BoardConfig.mk b/sabresd_6sx/BoardConfig.mk
index 91e30c5..2cbe642 100644
--- a/sabresd_6sx/BoardConfig.mk
+++ b/sabresd_6sx/BoardConfig.mk
@@ -125,5 +125,6 @@ BOARD_SEPOLICY_UNION := \
        rild.te \
        init.te \
        netd.te \
-       bootanim.te
+       bootanim.te \
+       dnsmasq.te
 
-- 
1.8.0

